File: string-combo/JavaScript/string-combo.js
// Write a JavaScript function that generates all combinations of a string.
// Example string : 'dog'
// Expected Output : d,do,dog,o,og,g

function comboCount(string){
  var arr = string.split('')
  arr.forEach(function(entry) {
    var newArr = []
    var i = arr.indexOf(entry);
    while (i < arr.length) {
      newArr.push(arr[i])
      console.log(newArr.join(""));
      i++;
    };
  })
};


comboCount("dog");

File: string-combo/Ruby/string-combo.rb
# // Write a JavaScript function that generates all combinations of a string.
# // Example string : 'dog'
# // Expected Output : d,do,dog,o,og,g

class StringCombo

  def run(string)
    arr = string.split('')
    new_arr = []
    final_arr = []
    n = 0
    while n < arr.length
      new_arr.push(arr[n])
      final_arr.push(new_arr.join(''))
      n += 1
    end
    final_arr.join(', ')
  end
end
